manager/eris-mint/evm/snative.go

```diff
@@ -247,14 +247,27 @@ func (contract *SNativeContractDescription) Dispatch(appState AppState,
 	return function.F(appState, caller, remainingArgs, gas)
 }
 
-// We define the address of an SNative contact as the simplest possible hash of
-// its canonical name
-func (contract *SNativeContractDescription) Address() Word256 {
-	return LeftPadWord256([]byte(contract.Name))
+// We define the address of an SNative contact as the first 20 bytes of the sha3
+// hash of its name
+func (contract *SNativeContractDescription) Address() abi.Address {
+	var address abi.Address
+	copy(address[:], sha3.Sha3([]byte(contract.Name))[:abi.AddressLength])
```

Comment:
Ethereum takes the right most 20 bytes of the 256bit Keccak digest; not the first 20 bytes;

see eris-keys here https://github.com/eris-ltd/eris-keys/blob/master/crypto/key.go#L152-L153
or also here http://bitcoin.stackexchange.com/questions/42055/what-is-the-approach-to-calculate-an-ethereum-address-from-a-256-bit-private-key